STM32U0:  provide support for DAC peripherals

djiatsaf-st opened the pull request:
This PR adds digital to analog converter (DAC) peripherals  on the stm32U0  boards.

erwango commented on tests/drivers/dac/dac_api/src/test_dac.c:
```diff
@@ -4,107 +4,80 @@
-	defined(CONFIG_BOARD_NUCLEO_U5A5ZJ_Q) || \
-	defined(CONFIG_BOARD_NUCLEO_WL55JC) || \
+#if defined(CONFIG_BOARD_NUCLEO_F091RC) || defined(CONFIG_BOARD_NUCLEO_F207ZG) ||                  \
+	defined(CONFIG_BOARD_STM32F3_DISCO) || defined(CONFIG_BOARD_NUCLEO_F429ZI) ||              \
+	defined(CONFIG_BOARD_NUCLEO_F746ZG) || defined(CONFIG_BOARD_NUCLEO_F767ZI) ||              \
+	defined(CONFIG_BOARD_NUCLEO_F722ZE) || defined(CONFIG_BOARD_NUCLEO_G071RB) ||              \
+	defined(CONFIG_BOARD_NUCLEO_G431RB) || defined(CONFIG_BOARD_NUCLEO_G474RE) ||              \
+	defined(CONFIG_BOARD_NUCLEO_H743ZI) || defined(CONFIG_BOARD_NUCLEO_L073RZ) ||              \
+	defined(CONFIG_BOARD_NUCLEO_L152RE) || defined(CONFIG_BOARD_DISCO_L475_IOT1) ||            \
+	defined(CONFIG_BOARD_NUCLEO_L552ZE_Q) || defined(CONFIG_BOARD_STM32L562E_DK) ||            \
+	defined(CONFIG_BOARD_STM32H573I_DK) || defined(CONFIG_BOARD_B_U585I_IOT02A) ||             \
+	defined(CONFIG_BOARD_NUCLEO_U575ZI_Q) || defined(CONFIG_BOARD_NUCLEO_U5A5ZJ_Q) ||          \
+	defined(CONFIG_BOARD_NUCLEO_WL55JC) || defined(CONFIG_BOARD_NUCLEO_U083RC) ||              \
```
Clang-format is not always right

Please re-instantiate the previous format which was easier to read in this specific case.
For instance it was easy to see which board is added and exactly where it should be added.
With this format, adding one board will require to change all lines

Apply everywhere

erwango commented on boards/st/nucleo_u031r8/nucleo_u031r8.dts:
```diff
@@ -81,3 +81,9 @@
 	apb1-prescaler = <1>;
 };
+
+&dac1{
```
Please update board documentation ("supported features")

martinjaeger reviewed (commented):
Looks good, only got minor comments.

Can you please squash the last commit into the second one, so that the docs are updated together with the board `.dts` file?

djiatsaf-st committed:
dts: arm: st: u0: add dac node in dtsi file

all stm32u0 boards have only one and same
dac peripheral.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
boards: st: add dac node in dts file and update docs

-enable dac in nucleo_u031r8, stm32u083c_dk and nucleo_u083rc
- add dac as new supported interface in docs
- update yaml files

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
tests: drivers: dac: dac_api: src: update dac_api driver test

-add nucleo_u083rc and stm32u083c_dk  boards for testbench purpose.
- add min_ram property to skip test on boards with little ram size.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

erwango commented on boards/st/nucleo_u031r8/doc/index.rst:
Please update board's .yaml file

erwango reviewed (changes requested):
Last catch otherwise LGTM, thanks

erwango commented:
@djiatsaf-st To fix the issue of tests overflowing soc footprint, don't hesitate to add `    min_ram: xx` to impacted test cases yaml files. This will be helpful for others as well.

mathieuchopstm commented on boards/st/nucleo_u031r8/nucleo_u031r8.dts:
```diff
@@ -82,6 +82,12 @@
 	apb1-prescaler = <1>;
 };
 
+&dac1{
```
Nit:
```suggestion
&dac1 {
```

mathieuchopstm commented on boards/st/nucleo_u083rc/nucleo_u083rc.dts:
```diff
@@ -82,6 +82,13 @@
 };
 
+
+&dac1{
```
Nit:
```suggestion
&dac1 {
```

mathieuchopstm commented on boards/st/stm32u083c_dk/stm32u083c_dk.dts:
```diff
@@ -72,6 +72,12 @@
 	apb1-prescaler = <1>;
 };
 
+&dac1{
```
Nit:
```suggestion
&dac1 {
```

mathieuchopstm reviewed (dismissed):
LGTM. Minor non-blocking details:

martinjaeger commented on boards/st/nucleo_u083rc/nucleo_u083rc.dts:
```diff
@@ -82,6 +82,13 @@
 	apb1-prescaler = <1>;
 };
 
+
```
Unnecessary empty line.

martinjaeger reviewed (changes requested):
Now that @mathieuchopstm identified the missing spaces, let's get all minor style issues fixed for consistency. Afterwards it's definitely ready.
